src/bmconfigparser.py

```diff
@@ -270,3 +169,6 @@ def validate_bitmessagesettings_maxoutboundconnections(value):
         if value < 0 or value > 8:
             return False
         return True
+
+
+config = BMConfigParser()
```

Comment:
Another place with side effects is bad

Reply:
What alternative design do you propose? I think a module variable is an improvement singleton and I don't know other approaches.

Reply:
For me this approach seems worse than the singleton antipattern. Create the instance of `BMConfigParser()` upon program start and use it everywhere passing as argument to functions or accessing it through a slot of another object.

These changes also create potential conflicts all across the codebase. And current test coverage is definitely not enough for such big changes.

Reply:
but where to put it then? `state.py`?

Reply:
So you basically recommend dependency injection. Which is ok, I'm just not sure that can be done easily now because of legacy structures. Using a module variable is halfway there, we can do unit tests on isolated objects, and can be refactored into dependency injection later.

Reply:
It's not just tests though, there are some additions that prevent misbehaviour, like a clean reset, consolidation of default values and threads waiting for initialization. I'm not super happy that the commit is kind of big, just the old code is difficult to modify in small chunks. The commit makes the code easier to change in the future.